homeassistant/components/rejseplanen/coordinator.py

```diff
@@ -0,0 +1,174 @@
+"""Data update coordinator for Rejseplanen."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+
+from py_rejseplan.api.departures import DeparturesAPIClient
+from py_rejseplan.dataclasses.departure import Departure
+from py_rejseplan.dataclasses.departure_board import DepartureBoard
+from py_rejseplan.exceptions import APIError, ConnectionError, HTTPError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN, SCAN_INTERVAL_MINUTES
+
+_LOGGER = logging.getLogger(__name__)
+
+type RejseplanenConfigEntry = ConfigEntry[RejseplanenDataUpdateCoordinator]
+
+
+class RejseplanenDataUpdateCoordinator(DataUpdateCoordinator[DepartureBoard]):
+    """Class to manage fetching data from the Rejseplanen API."""
+
+    def __init__(
+        self, hass: HomeAssistant, config_entry: RejseplanenConfigEntry
+    ) -> None:
+        """Initialize."""
+
+        self.api = DeparturesAPIClient(auth_key=config_entry.data["api_key"])
+        self._stop_ids: set[int] = set()
+        self.last_update_success_time: datetime | None = None
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN} Data Update Coordinator",
+            update_interval=timedelta(minutes=SCAN_INTERVAL_MINUTES),
+            config_entry=config_entry,
+        )
+
+    async def _async_update_data(self) -> DepartureBoard:
+        """Update data via library."""
+        try:
+            board = await self.hass.async_add_executor_job(self._fetch_data)
+        except (
+            APIError,
+            HTTPError,
+        ) as error:  # runtime errors from the API
+            raise UpdateFailed(error) from error
+        except ConnectionError as error:  # network errors
+            _LOGGER.error("Connection error while fetching data: %s", error)
+            raise UpdateFailed(error) from error
+        except TypeError as error:
+            _LOGGER.error(
+                "Type error fetching data for stop %s: %s", self._stop_ids, error
+            )
+            raise UpdateFailed(error) from error
+        except Exception as error:  # Catch any other unexpected errors
+            _LOGGER.error(
+                "Unexpected error fetching data for stop %s: %s", self._stop_ids, error
+            )
+            raise UpdateFailed(error) from error
```

Comment:
Let's not log and raise, otherwise just add your user friendly error to the UpdateFailed one